merlin/systems/dag/ops/operator.py

```diff
@@ -41,22 +51,97 @@ def __repr__(self):
 
 
 class InferenceOperator(BaseOperator):
+    """
+    The basic building block of the Inference Node, the Inference Operator encapsulates
+    a transform with the appropriate functionality to handle all Ensemble graph
+    operations.
+    """
+
     @property
     def export_name(self):
+        """
+        Provides a clear common english identifier for this operator.
+
+        Returns
+        -------
+        String
+            Name of the current class as spelled in module.
+        """
         return self.__class__.__name__.lower()
 
     @abstractmethod
-    def export(self, export_path, input_schema, output_schema, node_id=None, version=1):
-        pass
+    def export(
+        self,
+        path: str,
+        input_schema: Schema,
+        output_schema: Schema,
+        params: dict = None,
+        node_id: int = None,
+        version: int = 1,
+    ) -> tuple(dict, list):
+        """
+        Export the class object as a config and all related files to the user defined path.
```

Comment:
s/user defined/user-defined/ because it's used as an adjective to modify the noun, path.